src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -114,16 +114,19 @@ protected override Task DisposeCoreAsync(bool initialized)
             }
         }
 
-        private async Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
+        private Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
         {
             IVsProjectRestoreInfo projectRestoreInfo = ProjectRestoreInfoBuilder.Build(sources.Item1);
 
             if (projectRestoreInfo != null)
             {
-                await _solutionRestoreService
-                    .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)
-                    .ConfigureAwait(false);
+                _projectVsServices.Project.Services.ProjectAsynchronousTasks
+                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(() => _solutionRestoreService
```

Comment:
Also, do we need to unregister the prior registered critical task here? I am wondering if we will end up queueing a whole bunch of critical tasks and then wait on all of them (some of them using stale projectRestoreInfo) when an explicit build is kicked off.